### PR TITLE
ci: Centralize pnpm installation and caching

### PR DESCRIPTION
**Pull request by tylerbutler**
We're now using pnpm in several different pipelines, so to ease maintenance, I created a re-usable template to install and configure pnpm. I replaced all individual installations of pnpm with this template.

I also set the task timeout for the pnpm cache to 5 minutes.

### CODE REVIEW

File: tools/pipelines/templates/build-npm-package.yml [hunk 1]
```diff
@@ -246,32 +246,17 @@ stages:
...
+            parameters:
+              buildDirectory: ${{ parameters.buildDirectory }}
 
         - task: Bash@3
           displayName: Install dependencies
           inputs:
             targetType: 'inline'
             workingDirectory: ${{ parameters.buildDirectory }}
             script: |
+              pwd
```
Comment (Josmithr) [inline tools/pipelines/templates/build-npm-package.yml:259]: Unintentional change?

File: tools/pipelines/build-protocol-definitions.yml [hunk 2]
```diff
@@ -41,6 +41,7 @@ trigger:
     - tools/pipelines/templates/build-npm-package.yml
     - tools/pipelines/templates/include-set-package-version.yml
     - tools/pipelines/templates/include-vars.yml
+    - tools/pipelines/templates/include-install-pnpm.yml
```
Comment (Josmithr) [inline tools/pipelines/build-protocol-definitions.yml:44]: (Not really for this PR, but a more general question) The list of templates we have to specify in our pipelines has gotten pretty large. Should we just include `tools/pipelines/templates` and eat the cost of potentially unnecessary runs to reduce the overhead / reduce the risk of missing things?

**Review by Josmithr — APPROVED** [review]

File: tools/pipelines/templates/build-npm-package.yml (hunk 1 above)
Comment (tylerbutler) [reply tools/pipelines/templates/build-npm-package.yml:259]: debugging leftovers

File: tools/pipelines/templates/include-install-pnpm.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      path: ${{ parameters.pnpmStorePath }}
+      restoreKeys: |
+        pnpm-store | "$(Agent.OS)"
+
+- task: Bash@3
+  displayName: Install and configure pnpm
+  inputs:
+    targetType: 'inline'
+    workingDirectory: ${{ parameters.buildDirectory }}
+    condition: or(succeeded(), canceled())
```
Comment (Josmithr) [inline tools/pipelines/templates/include-install-pnpm.yml:41]: Nit: Might be worth a brief comment explaining why we want to continue on canceled here.

**Pull request merged**